### ISSUE TITLE
How can I get UserCredential ?

### ISSUE THREAD

**Issue by alexnu** [firebase/firebaseui-web#348]
After sign up when the user logs in for the first time, I want to create a new profile and save it to the database. Is there a way to get `UserCredential` so I can check`isNewUser()`?

Thanks

**Comment by wti806** [comment]
@alexnu We are likely to return UserCredential in signInSuccess callback in next release. Currently you can compare the creationTime and lastSignInTime from user metadata to tell if a user is newly created.

**Comment by wti806** [comment]
Personally I suggest you to wait for our next release, it should be really soon. I tested the lastSignInTime field and it works as expected. How did you get the user reference to check this field? In SignInSuccess callback or from onAuthStateChanged? And how do you trigger your profile creation?

**Comment by alexnu** [comment]
Okay, thanks for your suggestion. I tried both SignInSuccess callback and onAuthStateChanged and the result is the same. My profile creation is triggered by this condition:

```
if (user.metadata.creationTime === user.metadata.lastSignInTime) {
// create profile
}
```

**Comment by bojeil-google** [comment]
I actually think this may not work in our case, as we sign in twice (once internally and again on the external Auth instance). I suggest you wait for this to be implemented on our end (we are in the middle of this implementation) or just basically add the user profile to db if it doesn't already exist. You basically check if the user exists in your db (existing user), if not add it and consider it a new user.

**Comment by alexnu** [comment]
Okay I've added an extra check for if profile has already been created and will replace it once this feature is completed.

Thanks for you help.

**Comment by bojeil-google** [comment]
Thanks @alexnu. This should be done soon. We'll keep you posted.

**Comment by wti806** [comment]
@alexnu  Hi, Alex. We just released 2.7.0 which includes a new callback `signInSuccessWithAuthResult`. You can switch to the new callback and check `authResult.additionalInfo.isNewUser` field for your use case. Let me know if you have any questions. Thanks

**Comment by alexnu** [comment]
Thanks @wti806, I already tried it and it works great!
